src/main/java/org/kohsuke/github/GitHub.java

```diff
@@ -287,17 +295,61 @@ public void setConnector(HttpConnector connector) {
      */
     public GHRateLimit getRateLimit() throws IOException {
         try {
-            return retrieve().to("/rate_limit", JsonRateLimit.class).rate;
+            return rateLimit = retrieve().to("/rate_limit", JsonRateLimit.class).rate;
         } catch (FileNotFoundException e) {
             // GitHub Enterprise doesn't have the rate limit, so in that case
             // return some big number that's not too big.
             // see issue #78
             GHRateLimit r = new GHRateLimit();
             r.limit = r.remaining = 1000000;
-            long hours = 1000L * 60 * 60;
-            r.reset = new Date(System.currentTimeMillis() + 1 * hours );
-            return r;
+            long hour = 60L * 60L; // this is madness, storing the date as seconds in a Date object
+            r.reset = new Date((System.currentTimeMillis() + hour) / 1000L );
```

Comment:
I am sorry, but what the crazy calculation do you do here?

hour - value in second
System.currentTimeMillis() - return millisecond

(millisecond + second)/ 1000L - strange unpredictable type of second, we can say this type of seconds as X.

By javadoc:
public Date(long date)
date   the milliseconds since January 1, 1970, 00:00:00 GMT.

if we will use new Date( X ), then we can't predict what type of date we will receive.